File: js/pigLatin.js
exports.translate = function (word) {
  let wordList = word.split(" ");
  for (let index in wordList) {
    temp = wordList[index];
    if (temp[0].match(/[aeiou]/i)) {
      temp += "ay";
    } else {
      let vowelIndex = temp.indexOf(temp.match(/[aeio]/i));
      if (temp.match(/[!,.?;:]/)) {
        temp =
          temp.slice(vowelIndex, temp.length - 1) +
          temp.slice(0, vowelIndex) +
          "ay" +
          temp.slice(temp.length - 1);
      } else {
        temp = temp.slice(vowelIndex) + temp.slice(0, vowelIndex) + "ay";
      }
      if (wordList[index][0].match(/[A-Z]/)) {
        temp = temp[0].toUpperCase() + temp.slice(1).toLowerCase();
      }
    }
    wordList[index] = temp;
  }
  return wordList.join(" ");
};

File: python/pig_latin.py
import re

def translate(word_or_phrase):
  word_list = word_or_phrase.split(" ")
  for i in range(len(word_list)):
    temp = word_list[i]
    if re.search("[aeiou]", temp[0]):
      temp += 'ay'
    else:
      vowelIndex = re.search("[aeio]", temp).start()
      if re.search("[!,.?;:]", temp):
        temp = temp[vowelIndex:len(temp)-1]+temp[0:vowelIndex]+'ay'+temp[len(temp)-1]
      else:
        temp = temp[vowelIndex:]+temp[0:vowelIndex]+'ay'
      if re.search("[A-Z]", word_list[i][0]):
        temp = temp[0].upper()+temp[1:].lower()
    word_list[i] = temp
  return " ".join(word_list)